autoreject/autoreject.py

```diff
@@ -215,8 +215,10 @@ def get_rejection_threshold(epochs, decim=1, random_state=None,
     if decim > 1:
         epochs = epochs.copy()
         epochs.decimate(decim=decim)
-
-    cv = KFold(n_splits=cv, random_state=random_state)
+    
+    if isinstance(cv, int):
+        cv = KFold(n_splits=cv, random_state=random_state)
```

Comment:
maybe use this: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.check_cv.html to check that the `cv` instance makes sense?

And then you need to update the docstring above to say you accept any `cv` object from `scikit-learn`